include/pmacc/eventSystem/events/kernelEvents.hpp

```diff
@@ -44,124 +44,175 @@
 #endif
 
 
-namespace pmacc
+namespace pmacc::exec
 {
-    namespace exec
+    namespace detail
     {
-        /** configured kernel object
+        /** Kernel with dynamic shared memory
          *
-         * this objects contains the functor and the starting parameter
+         * This implements the possibility to define dynamic shared memory without
+         * specializing the needed alpaka trait BlockSharedMemDynSizeBytes for each kernel with shared memory.
+         * The trait BlockSharedMemDynSizeBytes is defined by PMacc for all types of KernelWithDynSharedMem.
+         */
+        template<typename T_Kernel>
+        struct KernelWithDynSharedMem : public T_Kernel
+        {
+            size_t const m_dynSharedMemBytes;
+
+            KernelWithDynSharedMem(T_Kernel const& kernel, size_t const& dynSharedMemBytes)
+                : T_Kernel(kernel)
+                , m_dynSharedMemBytes(dynSharedMemBytes)
+            {
+            }
+        };
+
+        //! Mata data of a device kernel
+        class KernelMetaData
+        {
+            //! file name
+            std::string const m_file;
+            //! line number
+            size_t const m_line;
+
+        public:
+            KernelMetaData(std::string const& file, size_t const line) : m_file(file), m_line(line)
+            {
+            }
+
+            //! file name from where the kernel is called
+            std::string getFile() const
+            {
+                return m_file;
+            }
+
+            //! line number in the file where the kernel is called
+            size_t getLine() const
+            {
+                return m_line;
+            }
+        };
+
+
+        /** Configured kernel object.
+         *
+         * This objects contains the kernel functor, kernel meta information and the starting parameters.
+         * Object is used to enqueue the kernel with user parameters to the device.
          *
          * @tparam T_Kernel pmacc Kernel object
-         * @tparam T_VectorGrid type which defines the grid extents (type must be castable to cupla dim3)
-         * @tparam T_VectorBlock type which defines the block extents (type must be castable to cupla dim3)
          */
-        template<typename T_Kernel, typename T_VectorGrid, typename T_VectorBlock>
+        template<typename T_Kernel>
         struct KernelStarter;
 
-        /** wrapper for the user kernel functor
+        /** Wrapper for the user kernel functor.
```

Comment:
This not a sentence: the verb is missing.
```suggestion
        /** wrapper for the user kernel functor
```